### PR TITLE
KAFKA-2558: ServerShutdownTest is failing intermittently

### PR DESCRIPTION
**Pull request by fpj**
See jira for a description.

### CODE REVIEW

**Comment by onurkaraman** [comment]
LGTM.

I ran into this issue yesterday:
https://issues.apache.org/jira/browse/KAFKA-2438?page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel&focusedCommentId=14876573#comment-14876573

**Comment by fpj** [comment]
Thanks for checking @onurkaraman

File: core/src/main/scala/kafka/controller/ControllerChannelManager.scala [hunk 1]
```diff
@@ -36,7 +36,7 @@ import kafka.common.{KafkaException, TopicAndPartition}
 import collection.Set
 import collection.JavaConverters._
 
-class ControllerChannelManager(controllerContext: ControllerContext, config: KafkaConfig, time: Time, metrics: Metrics) extends Logging {
+class ControllerChannelManager(controllerContext: ControllerContext, config: KafkaConfig, time: Time, metrics: Metrics, testName: Option[String] = None) extends Logging {
```
Comment (ijuma) [inline core/src/main/scala/kafka/controller/ControllerChannelManager.scala]: Passing a parameter called `testName` to a non test class is not ideal. We could either override the thread creation in the test or pass a more generic parameter (maybe requestThreadNamePrefix). What do you think?

Comment (fpj) [inline core/src/main/scala/kafka/controller/ControllerChannelManager.scala]: I'm fine with making the name more generic (e.g., threadNamePrefix).

**Comment by guozhangwang** [comment]
Could you fix TestUtils.verifyNonDaemonThreadsStatus also?

File: core/src/main/scala/kafka/controller/ControllerChannelManager.scala (hunk 1 above)
Comment (guozhangwang) [inline core/src/main/scala/kafka/controller/ControllerChannelManager.scala]: I think threadNamePrefix is fine.

**Comment by guozhangwang** [comment]
LGTM.

**Comment by b-goyal** [comment]
See jira for a description.

**Pull request closed without merging**